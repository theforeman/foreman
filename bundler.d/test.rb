group :test do
  gem 'mocha', '~> 1.1', :require => false
  gem 'simplecov', '~> 0.9'
  gem 'spork-minitest', '0.0.3'
  gem 'single_test', '~> 0.6'
  gem 'minitest', '5.1.0'
  gem 'minitest-spec-rails' #, '~> 4.7'
  #gem 'ci_reporter', '2.0.0', :require => false
  gem 'ci_reporter_minitest', :require => false
  gem 'capybara', '~> 2.0'
  gem 'database_cleaner', '~> 1.3'
  gem 'launchy', '~> 2.4'
  gem 'spork', '~> 0.9'
  gem 'factory_girl_rails', '~> 4.5', :require => false
  gem 'rubocop-checkstyle_formatter', '~> 0.1'
  gem "poltergeist"
end
