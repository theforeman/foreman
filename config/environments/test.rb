# Track deprecation warnings in test environment as early as possible, but pause processing of
# deprecations until all plugins are registered (prior to the finisher_hook initializer) to ensure
# the whitelist is fully configured. This is done in the after_initialize block below.
ASDeprecationTracker.pause!

Foreman::Application.configure do
  # Settings specified here will take precedence over those in config/application.rb.

  # The test environment is used exclusively to run your application's
  # test suite. You never need to work with it otherwise. Remember that
  # your test database is "scratch space" for the test suite and is wiped
  # and recreated between test runs. Don't rely on the data there!
  config.cache_classes = true

  # Do not eager load code on boot. This avoids loading your whole application
  # just for the purpose of running a single test. If you are using a tool that
  # preloads Rails for running tests, you may have to set it to true.
  config.eager_load = false

  # Configure static asset server for tests with Cache-Control for performance.
  config.serve_static_files   = true
  config.static_cache_control = 'public, max-age=3600'

  # Adds additional error checking when serving assets at runtime.
  # Checks for improperly declared sprockets dependencies.
  # Raises helpful error messages.
  config.assets.raise_runtime_errors = true

  # Show full error reports and disable caching.
  config.consider_all_requests_local       = true
  config.action_controller.perform_caching = false

  # Raise exceptions instead of rendering exception templates.
  config.action_dispatch.show_exceptions = false

  # Disable request forgery protection in test environment.
  config.action_controller.allow_forgery_protection = false

  # Tell Action Mailer not to deliver emails to the real world.
  # The :test delivery method accumulates sent emails in the
  # ActionMailer::Base.deliveries array.
  config.action_mailer.delivery_method = :test
  config.action_mailer.perform_deliveries = true
  ActionMailer::Base.deliveries.clear

  # Use SQL instead of Active Record's schema dumper when creating the test database.
  # This is necessary if your schema can't be completely dumped by the schema dumper,
  # like if you have constraints or database-specific column types
  # config.active_record.schema_format = :sql

  # Should ANSI color codes be used when logging information
  config.colorize_logging = Foreman::Logging.config[:colorize]

  # Print deprecation notices to the stderr.
  config.active_support.deprecation = :stderr

  # Raise exception on mass assignment of unfiltered parameters
  config.action_controller.action_on_unpermitted_parameters = :strict

  # Use separate cache stores for parallel_tests
  config.cache_store = :file_store, Rails.root.join("tmp", "cache", "paralleltests#{ENV['TEST_ENV_NUMBER']}")

  # Enable automatic creation/migration of the test DB when running tests
  config.active_record.maintain_test_schema = true

  # Randomize the order test cases are executed.
  config.active_support.test_order = :random

  config.webpack.dev_server.enabled = false

  # Whitelist all plugin engines by default from raising errors on deprecation warnings for
  # compatibility, allow them to override it by adding an ASDT configuration file.
  config.after_initialize do
    Foreman::Plugin.all.each do |plugin|
      unless File.exist?(File.join(plugin.path, 'config', 'as_deprecation_whitelist.yaml'))
        ASDeprecationTracker.whitelist.add(engine: plugin.id.to_s.gsub('-', '_'))
      end
    end
    ASDeprecationTracker.resume!
  end
end
