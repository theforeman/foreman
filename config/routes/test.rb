if Rails.env.test?
  Foreman::Application.routes.draw do
    resources :testable, :only => :index
    resources :testable_strong_params, :only => :create
    resources :my_hosts, :only => :create
    resources :testable_no_params, :only => :create

    namespace :api do
      namespace :v2 do
        resources :testable, :only => [:create, :index]
      end

      resources :testable, :only => :index do
        get :raise_error, :on => :collection
      end
    end
  end
end
