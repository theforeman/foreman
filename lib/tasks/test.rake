namespace :test do
  desc "Test API"
  Rake::TestTask.new(:api) do |t|
    t.libs << "test"
    t.pattern = 'test/functional/api/**/*_test.rb'
    t.verbose = true
  end
end

namespace :test do
  desc "Test lib source"
  Rake::TestTask.new(:lib) do |t|
    t.libs << "test"
    t.pattern = 'test/lib/**/*_test.rb'
    t.verbose = true
  end
end

Rake::Task[:test].enhance do
  Rake::Task['test:lib'].invoke
end
