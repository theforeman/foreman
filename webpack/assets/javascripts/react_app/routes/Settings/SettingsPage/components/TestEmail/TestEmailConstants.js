export const TEST_EMAIL_REQUEST = 'TEST_EMAIL_REQUEST';
export const TEST_EMAIL_RESPONSE = 'TEST_EMAIL_RESPONSE';
export const TEST_EMAIL_URL = '/users/:id/test_mail';
