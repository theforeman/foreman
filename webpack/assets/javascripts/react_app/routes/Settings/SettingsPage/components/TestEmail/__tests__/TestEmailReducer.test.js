import { testReducerSnapshotWithFixtures } from '@theforeman/test';

import { TEST_EMAIL_REQUEST, TEST_EMAIL_RESPONSE } from '../TestEmailConstants';

import reducer from '../TestEmailReducer';

const fixtures = {
  'should return initial state': {},
  'should return email loading': {
    action: {
      type: TEST_EMAIL_REQUEST,
    },
  },
  'should return email stopped loading': {
    action: {
      type: TEST_EMAIL_RESPONSE,
    },
  },
};

describe('TestEmailReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
