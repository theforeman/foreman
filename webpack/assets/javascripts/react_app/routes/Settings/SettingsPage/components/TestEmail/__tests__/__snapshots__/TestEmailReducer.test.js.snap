// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestEmailReducer should return email loading 1`] = `
Object {
  "loading": true,
}
`;

exports[`TestEmailReducer should return email stopped loading 1`] = `
Object {
  "loading": false,
}
`;

exports[`TestEmailReducer should return initial state 1`] = `
Object {
  "loading": false,
}
`;
